src/core/bonded_interactions/angle_cosine.hpp

```diff
@@ -74,21 +75,12 @@ calc_angle_cosine_3body_forces(Particle const *p_mid, Particle const *p_left,
  *  @param[out] f_right   Force on @p p_right.
  *  @retval false
  */
-inline bool calc_angle_cosine_force(Particle const *p_mid,
-                                    Particle const *p_left,
-                                    Particle const *p_right,
-                                    Bonded_ia_parameters const *iaparams,
-                                    double f_mid[3], double f_left[3],
-                                    double f_right[3]) {
-
-  Utils::Vector3d f_mid_v, f_left_v, f_right_v;
-  std::tie(f_mid_v, f_left_v, f_right_v) =
+inline bool calc_angle_cosine_force(
+    Particle const *const p_mid, Particle const *const p_left,
+    Particle const *const p_right, Bonded_ia_parameters const *const iaparams,
+    Utils::Vector3d &f_mid, Utils::Vector3d &f_left, Utils::Vector3d &f_right) {
```

Comment:
those output parameters are evil

Reply:
agreed, they have already been dealt with in [fix-2502-p2](https://github.com/jngrad/espresso/compare/fix-2502-p1b...fix-2502-p2#diff-94e6cd4fc1e0b199e0902e54d014c802)